local/compose.go

```diff
@@ -59,7 +59,7 @@ func (s *composeService) Up(ctx context.Context, project *types.Project, detach
 		return err
 	}
 	for k, network := range project.Networks {
-		if !network.External.External && network.Name != "" {
+		if !network.External.External && network.Name == k {
```

Comment:
ATM we can't tell if the network name was set by the user or set by default as the key value. Assuming if name = key that the name was not set by user. 
Will do a separate PR to update compose-go and let network name empty if not set by user, so we can differentiate the 2 cases